src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -59,14 +59,20 @@ class DeltaAnalysis(session: SparkSession, conf: SQLConf)
       }
 
     // INSERT OVERWRITE by ordinal
-    case a @ OverwriteByExpression(
-        DataSourceV2Relation(d: DeltaTableV2, _, _, _, _), _, query, _, false)
+    case o @ OverwriteByExpression(
+        DataSourceV2Relation(d: DeltaTableV2, _, _, _, _), deleteExpr, query, _, false)
       if query.resolved && needsSchemaAdjustment(d.name(), query, d.schema()) =>
       val projection = normalizeQueryColumns(query, d)
       if (projection != query) {
-        a.copy(query = projection)
+        val aliases = AttributeMap(projection.projectList.collect {
```

Comment:
nit: `projection.output.zip(query.output).filter { case (l, r) => !l.sameRef(r) }`

Reply:
then we don't need to change return type of normalizeQueryColumns

Reply:
Sure. Updated.

Reply:
@cloud-fan ut passed